tests/base_scraper/test_base_scraper.py

```diff
@@ -5,133 +5,120 @@
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import os
-import unittest
-
-import mozhttpd_base_test as mhttpd
+import pytest
 import requests
 
 import mozdownload
 import mozdownload.errors as errors
 from mozdownload.scraper import PLATFORM_FRAGMENTS
 from mozdownload.utils import create_md5, urljoin
 
-
-class TestBaseScraper(mhttpd.MozHttpdBaseTest):
-    """Testing the basic functionality of the Base Scraper Class"""
-
-    def test_platform_regex(self):
-        """Test for correct platform_regex output"""
-
-        for key in PLATFORM_FRAGMENTS:
-            scraper = mozdownload.Scraper(destination=self.temp_dir,
-                                          platform=key)
-            self.assertEqual(scraper.platform_regex,
-                             PLATFORM_FRAGMENTS[key])
-
-    def test_download(self):
-        """Test download method"""
-
-        filename = 'download_test.txt'
-        # standard download
-        test_url = urljoin(self.wdir, filename)
-        scraper = mozdownload.DirectScraper(url=test_url,
-                                            destination=self.temp_dir,
-                                            logger=self.logger)
+@pytest.mark.parametrize('platform_key,platform_value', PLATFORM_FRAGMENTS.items())
+def test_platform_regex(tmpdir, platform_key, platform_value):
+    """Test for correct platform_regex output"""
+    scraper = mozdownload.Scraper(destination=str(tmpdir), platform=platform_key)
+    assert scraper.platform_regex == platform_value
+
+def test_standard_download(httpd, tmpdir):
+    """Test standard download method"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    scraper.download()
+    assert os.path.isfile(os.path.join(str(tmpdir), filename))
+
+def test_compare_download(httpd, tmpdir):
+    """Compare original and downloaded file via md5 hash"""
+    filename = 'download_test.txt'
+    test_url = urljoin(httpd.get_url(), filename)
+    scraper = mozdownload.DirectScraper(url=test_url, destination=str(tmpdir))
+    scraper.download()
+    md5_original = create_md5(os.path.join(httpd.router.doc_root, filename))
+    md5_downloaded = create_md5(os.path.join(str(tmpdir), filename))
+    assert md5_original == md5_downloaded
+
+def test_request_exception(httpd, tmpdir):
+    """RequestException"""
+    test_url1 = urljoin(httpd.get_url(), 'does_not_exist.html')
+    scraper1 = mozdownload.DirectScraper(url=test_url1, destination=str(tmpdir))
+    with pytest.raises(requests.exceptions.RequestException):
```

Comment:
Actually I wonder why we don't raise a `NotFoundError` here, which would be more appropriate. Mind filing a new issue about that?

Reply:
I tried using errors.NotFoundError from mozdownload.errors but it resulted in failing of the test. I am guessing it is due to exception raised by scraper.download - https://github.com/mozilla/mozdownload/blob/master/mozdownload/scraper.py#L319. This inference could be wrong.

Reply:
As the issue is created separately for this and dedicatedly in milestone so this PR handles what was asked. I wish to take issue 534 if you guide me through it based on the comments mentioned above.

Reply:
Sure. We can follow-up on that other issue. I'm resolving this one now.